stubs/CoreImmutableClasses.phpstub

```diff
@@ -1,11 +1,106 @@
 <?php
 
-/**
- * @psalm-immutable
- */
 class DateTimeImmutable implements DateTimeInterface
 {
     public function __construct(string $datetime = "now", DateTimeZone $timezone = null) {}
+
+    /**
+     * @psalm-pure
```

Comment:
I think these should all be `@psalm-mutation-free` instead of `@psalm-pure`. Compare: https://psalm.dev/r/39eb3a7c35 https://psalm.dev/r/7a678770a4.

Reply:
from a caller perspective, these should be usable in a pure context, and `$this` should still be immutable-ish?

Reply:
Mutation free methods are usable from a pure context: https://psalm.dev/r/1ac851b862

I'm not really sure if mutation-free vs pure is a useful distinction for us to have, but that's the way it is right now :shrug:

Reply:
As long as it works, I'm fine with going either way.

Should I adjust it?

Reply:
I'd say so, I'm not sure if we have any logic that cares about the difference between mutation-free and pure, but better safe than sorry.

Reply:
68978b9e19c70036e6e5077a6234af513afdc77f